client/src/pages/SignUpModal.tsx

```diff
@@ -100,52 +109,70 @@ const SignUpModal: FunctionComponent = () => {
         </Popover>
     );
-    if (redirectToHome){
-        return (<Redirect to="/test"/>)
+    if (redirect !== "") {
+      return <Redirect to={redirect} />;
     }
     return (
         <CommonModal title={modalTitle} subtitle={subtitle} show={show} handleClose={handleClose} body={
             <div>
                 <form onSubmit={handleClick}>
-                
                 <div>
-                    <div className="text signup">
-                        Email address
-                    </div>
+                <div className="row">
+                  <div className="text signup">Email Address</div>
+                  <div className="text error">{errors.email}</div>
+                </div>
+                <input
+                  name="email"
+                  placeholder="Enter your company email"
+                  type="text"
+                  value={email}
+                  className={errors.email ? "input-field error" : "input-field"}
+                  onChange={onChangeEmail}
+                />
+              </div>
+              <div>
+                <div className="row">
+                  <div className="text signup">
+                    Password<div>{errors.password}</div>
+                  </div>
+                </div>
+
+                <div className="pass-req">
+                  <OverlayTrigger placement="bottom" overlay={popover}>
                     <input
-                        name="email"
-                        placeholder="Enter your company email"
-                        type="text"
-                        value={email}
-                        className="input-field"
-                        onChange={onChangeEmail}
+                      type="password"
+                      name="password"
+                      className={
+                        errors.password ? "input-field error" : "input-field"
+                      }
+                      placeholder="Enter your password"
+                      value={password}
+                      onChange={onChangePass}
                     />
+                  </OverlayTrigger>
                 </div>
-                <div>
-                    <div className="text signup">
-                        Password
-                    </div>
-                
-                    <div className="pass-req">
-                        <OverlayTrigger placement="bottom" overlay={popover}>
-                            <input
-                                type="password"
-                                name="password"
-                                className="input-field"
-                                placeholder="Enter your password"
-                                value={password}
-                                onChange={onChangePass}           
-                            />
-                        </OverlayTrigger>
-                    </div>
+              </div>
+              <div style={{ height: 40 }} />
```

Comment:
No inline styling please :)))

Reply:
I'm thinking of just making an atom called Spacer that takes in height as a prop and returns the corresponding div with that height - a trick from my first co-op lol

Reply:
lol i don't think i've heard of that trick yet
Would it be possible to just change the margins on the surrounding elements instead of adding a spacer?

Reply:
What I'm thinking though is that we want our components and component styling to be reusable. If we want to include an input field or button in another component, we'd want it to have minimal margin/padding attached to it as possible because adding spacing like this is a lot easier than having to end up doing workarounds in another component for margins/padding that are only there because it was convenient for one use of the component or having to do unique styling for every modal for atoms and molecules that are the same. What do you think?